### PR TITLE
In sharing panel, when the user is an owner of a resource, and wants …

### PR DESCRIPTION
**Pull request by engineSound**
#2127 

### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [x] Peer Code review and approval

### Positive Test Case
1. Only when you are a logged in user, you are an owner of a resource, you try to reduce owner to "view" or "edit" mode, you get a warning. Other than this case, no warning dialog.

### CODE REVIEW

File: theme/templates/resource-landing-page/modals.html [hunk 1]
```diff
@@ -1946,6 +1946,15 @@ <h3 class="panel-title">Sharing status</h3>
             </p>
         </div>
     {% endblock %}
+
+    {% block change-share-permission %}
+        <div id="dialog-confirm-change-share-permission" title="Warning! Loss of owner privilege" style="display:none">
+            <p><span class="ui-icon ui-icon-alert" style="float:left; margin:12px 12px 50px 0;"></span>
+                This action will make you lose owner privilege to this resource. You will no longer be an owner
```
Comment (Maurier) [inline theme/templates/resource-landing-page/modals.html]: Update this text as recommended here: https://github.com/hydroshare/hydroshare/issues/2127#issuecomment-459900604

Comment (engineSound) [reply theme/templates/resource-landing-page/modals.html]: Will do.

**Comment by engineSound** [comment]
I felt the prevention should come from backend. I clearly see we are doing the business logic on behalf of backend. @Maurier What do you think?

**Comment by Maurier** [comment]
@engineSound For this case, I don't think we should handle these notifications from back end. This notification concerns only the web client and we should handle it from the front end as to avoid adding unnecessary requests.

File: theme/static/js/resourceLandingAjax.js [hunk 2]
```diff
@@ -413,7 +413,67 @@ function promptSelfRemovingAccess(form_id){
...
+
+    if (currentUserID == userID
+        && previousAccess == "Is owner"
+        && previousAccess != clickedAccess){
+         return REQUIRED;
+    }
+    return NOT_REQUIRED;
+}
+function change_share_permission_ajax_submit(form_id, previousAccess, clickedAccess) {
+    if (!isSharePermissionPromptRequired(form_id, previousAccess, clickedAccess)) {
```
[the remark this answers is not in the document]
Comment (engineSound) [reply theme/static/js/resourceLandingAjax.js]: I reproduced the scenario you described. When inviting a person who is already in the list of people who have given shared permission. In this case, the person should not be invited. For such case, when the “Add” button in the invite section is clicked, it is more appropriate to have a dialog stating the user to add is already in the sharing user list.

[the remark this answers is not in the document]
Comment (Maurier) [reply theme/static/js/resourceLandingAjax.js]: @engineSound, I guess I'm ok with a warning like this, but I would check with @horsburgh or @dtarb before we merge this change.

File: theme/templates/resource-landing-page/modals.html [hunk 3]
```diff
@@ -1937,15 +1937,26 @@ <h3 class="panel-title">Sharing status</h3>
...
-    {% endblock %}
+    <div id="dialog-confirm-delete-self-access" class="access-dialog-title" title="Warning! Loss of access"> 
+        <p><span class="ui-icon ui-icon-alert access-warning"></span>
+            This action will remove your access to this resource and you will no longer be able to
+            access it without someone else sharing it with you again. This is not reversible,
+            so <strong>Remove</strong> with caution.
+        </p>
+    </div>
+    
+    <div id="dialog-confirm-change-share-permission" class="access-dialog-title" title="Warning! Loss of owner access">
```
Comment (Maurier) [inline theme/templates/resource-landing-page/modals.html]: For this warning, it's better to show it without a pop up. For consistency, we try to use this feature as confirmation dialogs. Please use this code instead to show this warning:
```
$("#div-invite-people").find(".label-danger").remove(); // Remove previous alerts
$("#div-invite-people").append("<span class='label label-danger'><strong>Failed to add access: </strong>The selected user already has access.</span>");
```

File: theme/templates/resource-landing-page/modals.html [hunk 4]
```diff
@@ -1937,15 +1937,26 @@ <h3 class="panel-title">Sharing status</h3>
...
+        <p><span class="ui-icon ui-icon-alert access-warning"></span>
+        This action will remove your owner access to this resource. This removal is not reversible. Owner 
+        access can only be retrieved by another owner later granting you owner access. <strong>Remove</strong> with caution.
+        </p>
+    </div>
+
+    <div id="dialog-user-invited" class="access-dialog-title" title="Information: Access">
```
Comment (Maurier) [inline theme/templates/resource-landing-page/modals.html]: For this warning, it's better to show it without a pop up. For consistency, we try to use this feature as confirmation dialogs. Please use this code instead to show this warning:

```
$("#div-invite-people").find(".label-danger").remove(); // Remove previous alerts
$("#div-invite-people").append("<span class='label label-danger'><strong>Failed to add access: </strong>The selected user already has access.</span>");
```

Comment (engineSound) [reply theme/templates/resource-landing-page/modals.html]: I like it!

**Comment by engineSound** [comment]
test this please

**Comment by engineSound** [comment]
retest this, please

**Comment by engineSound** [comment]
retest this, please

**Comment by phuongdm** [comment]
retest this, please

**Comment by phuongdm** [comment]
retest this, please

**Comment by phuongdm** [comment]
retest this, please

**Comment by phuongdm** [comment]
retest this, please

**Comment by cuahsimarko** [comment]
@engineSound Can you maybe close this out?

**Comment by engineSound** [comment]
@Maurier Hi Mauriel, can you approve this?  I made changes per your request. @cuahsimarko @martinseul both asked me for update.

**Pull request merged**